code/modules/mechanics/MechanicMadness.dm

```diff
@@ -1820,6 +1820,187 @@
+		//...FIFO queue with clobbering like an audio buffer
+
+/obj/item/mechanics/buffercomp
+	name = "Buffer Component"
+	desc = ""
+	icon_state = "comp_buffer"
+	cooldown_time = 0.4 SECONDS
+	var/list/buffer = list()
+	var/buffer_size = 15
+	//I wanted to set this to the same limit as the selection component
+	//But the selection compoment doesn't actually have a limit in the additem() proc
+	//It kind of worries me
+	var/buffer_max_size = 200
+	//Ring reader ring writer
+	var/rr = 1
+	var/rw = 1
+
+	var/buffer_model = RING_BUFFER
+	var/buffer_string = "ring"
+	var/buffer_desc = "This mode outputs from oldest to newest, overwriting the oldest signal when full."
+	var/changesig = 0
+
+	get_desc()
+		. += "<br>[SPAN_NOTICE("Delay is [cooldown_time] (in 10ths of a second). <br> Buffer size is [buffer_size].\
+		 <br> Buffer mode is [buffer_string].<br>[buffer_desc]")]"
+
+	New()
+		..()
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"buffer", PROC_REF(buffer))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Delay",PROC_REF(setDelay))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Buffer Mode",PROC_REF(setModel))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Buffer Size",PROC_REF(setBufferSize))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ALLOW_MANUAL_SIGNAL)
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Toggle Signal Changing",PROC_REF(toggleDefault))
+
+
+
+	proc/setModel(obj/item/W as obj, mob/user as mob)
+		var/model = input("Set the buffer mode to what?", "Mode Selector",buffer_string) in list("FIFO","FILO","ring","random")
+		if(model == "FIFO")
+			buffer_desc = "This mode outputs from oldest to newest, dropping signals when full."
+			buffer_model = FIFO_BUFFER
+		if(model == "FILO")
+			buffer_desc = "This mode outputs from newest to oldest, dropping signals when full."
+			buffer_model = FILO_BUFFER
+		if(model == "ring")
+			buffer_desc = "This mode outputs from oldest to newest, overwriting the oldest signal when full."
+			buffer_model = RING_BUFFER
+		if(model == "random")
+			buffer_desc = "This mode outputs signals randomly, randomly overwriting previous signals when full."
+			buffer_model = RANDOM_BUFFER
+		buffer_string = model
+		boutput(user, "You set the buffer mode to [model]")
+		tooltip_rebuild = 1
+		qdel(buffer)
+		buffer = list()
+		return 1
+
+	//Thanks delay component
+	proc/setDelay(obj/item/W as obj, mob/user as mob)
+		var/inp = input(user, "Enter delay in 10ths of a second:", "Set delay", cooldown_time) as num
+		if(!in_interact_range(src, user) || user.stat)
+			return 0
+		inp = min(inp, 60)
+		inp = max(4, inp)
+		if(!isnull(inp))
+			cooldown_time = inp
+			tooltip_rebuild = 1
+			boutput(user, "Set delay to [inp]")
+			return 1
+		return 0
+
+
+	proc/setBufferSize(obj/item/W as obj, mob/user as mob)
+		var/inp = input(user,"Set size of signal buffer","Buffer size", buffer_size) as num
+		if(!in_interact_range(src, user) || user.stat)
+			return 0
```

Comment:
not a big fan of this input checking, there's much better methods these days. For example, if a user is paralyzed on the ground this would still work. try `can_act` and `tgui_input_number`